src/app/legacy/components/Footer/index.jsx

```diff
@@ -70,24 +70,51 @@ const StyledAmpCookieSettingsButton = styled(AmpCookieSettingsButton)`
   }
 `;
 
+const openPrivacyManagerModal = e => {
+  e.preventDefault();
+  if (window.dotcom && window.dotcom.openPrivacyManagerModal) {
+    window.dotcom.openPrivacyManagerModal();
+  }
+};
+
 const SitewideLinks = ({
   links,
   trustProjectLink,
   copyrightText,
   externalLink,
   isAmp,
+  showAdsBasedOnLocation,
   script,
   service,
 }) => {
   const elements = links.map(({ id, text, href, lang }) => {
-    if (isAmp && id === 'COOKIE_SETTINGS') {
-      return (
-        <StyledAmpCookieSettingsButton lang={lang} service={service}>
-          {text}
-        </StyledAmpCookieSettingsButton>
-      );
+    if (id === 'COOKIE_SETTINGS') {
+      if (isAmp) {
+        return (
+          <StyledAmpCookieSettingsButton lang={lang} service={service}>
+            {text}
+          </StyledAmpCookieSettingsButton>
+        );
+      }
+
+      if (showAdsBasedOnLocation) {
+        const JSenabled = Number && typeof window !== 'undefined';
+        return (
+          <Link
+            service={service}
+            text={text}
+            href={href}
+            lang={lang}
+            onClick={openPrivacyManagerModal}
+            onlyShowIfJSenabled
+            visible={JSenabled}
```

Comment:
Can these values be combined? Looks like they kinda doing the same thing. If they are, you could use the `onClient()` function within `<Link />` itself to determine visibility?

Reply:
Thanks @amoore108 .
I've removed the 'visible' prop.